src/functions/locator.c

```diff
@@ -28,22 +30,17 @@
 
 #include <arpa/inet.h>
 
-static locator_t *g_locator = NULL;
+static locator_t g_locator = { 0 };
 
 
 locator_t *
 locator_new()
 {
-    if (g_locator != NULL)
-        return NULL;
-
-    g_locator = malloc(sizeof(locator_t));
-    /* test peer */
-    g_locator->peers_capacity = 64;
-    g_locator->peers = malloc(g_locator->peers_capacity * sizeof(peer_t));
-    g_locator->peers_size = 0;
+    g_locator.peers_capacity = 64;
+    g_locator.peers = malloc(g_locator.peers_capacity * sizeof(peer_t));
+    g_locator.peers_size = 0;
 
-    return g_locator;
```

Comment:
I don't think the new code is safe to call multiple times -- is this still supposed to be a singleton?